From b0eff200f9d00f34b705297ca10e7bfbc1d71276 Mon Sep 17 00:00:00 2001
From: Shiyong Li <a22381@motorola.com>
Date: Tue, 10 Nov 2015 14:07:49 -0800
Subject: [PATCH 457/959] IKKRNBSP-3493 arm/dts: 8952: BOE panel bring up

- BOE panel is supported on P1 or later.
- Vddio is connected to L6 without load switch on P1 or later.

Change-Id: Ia7b39b696c69eb1c2922c5e89479113d4fdf5694
Signed-off-by: Shiyong Li <a22381@motorola.com>
Reviewed-on: http://gerrit.mot.com/798205
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Shanshan Dai <daiss1@lenovo.com>
Reviewed-by: Jitao Jin <jinjt1@motorola.com>
Reviewed-by: Bang Nguyen <bangnguyen@motorola.com>
Reviewed-by: Joseph Swantek <jswantek@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 .../dsi-panel-mot-boe-546-1080p-cmd-common.dtsi    | 148 +++++++++++++++++++++
 .../dts/qcom/dsi-panel-mot-boe-546-1080p-cmd.dtsi  |  17 +++
 arch/arm/boot/dts/qcom/msm8952-athene-p0.dts       |   4 +
 arch/arm/boot/dts/qcom/msm8952-athene.dtsi         |   1 +
 4 files changed, 170 insertions(+)
 create mode 100644 arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd-common.dtsi
 create mode 100644 arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd.dtsi

diff --git a/arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd-common.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd-common.dtsi
new file mode 100644
index 00000000000..5bb8f0d1789
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd-common.dtsi
@@ -0,0 +1,148 @@
+/* Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+compatible = "qcom,mdss-dsi-panel";
+qcom,mdss-dsi-panel-name = "mipi_mot_cmd_boe_1080p_546_v0";
+qcom,mdss-dsi-panel-supplier = "boe";
+qcom,mdss-dsi-panel-type = "dsi_cmd_mode";
+qcom,mdss-dsi-panel-framerate = <60>;
+qcom,mdss-dsi-virtual-channel-id = <0>;
+qcom,mdss-dsi-stream = <0>;
+qcom,mdss-dsi-panel-width = <1080>;
+qcom,mdss-dsi-panel-height = <1920>;
+/* Todo: update dimension values based on BOE spec. */
+qcom,mdss-pan-physical-width-dimension = <68>;
+qcom,mdss-pan-physical-height-dimension = <121>;
+
+qcom,mdss-dsi-panel-clockrate = <807549120>;
+qcom,mdss-dsi-h-front-porch = <24>;
+qcom,mdss-dsi-h-back-porch = <40>;
+qcom,mdss-dsi-h-pulse-width = <4>;
+qcom,mdss-dsi-h-sync-skew = <0>;
+qcom,mdss-dsi-v-back-porch = <16>;
+qcom,mdss-dsi-v-front-porch = <16>;
+qcom,mdss-dsi-v-pulse-width = <2>;
+qcom,mdss-dsi-h-left-border = <0>;
+qcom,mdss-dsi-h-right-border = <0>;
+qcom,mdss-dsi-v-top-border = <0>;
+qcom,mdss-dsi-v-bottom-border = <0>;
+qcom,mdss-dsi-bpp = <24>;
+qcom,mdss-dsi-underflow-color = <0xff>;
+qcom,mdss-dsi-border-color = <0>;
+qcom,mdss-dsi-on-command = [
+	05 01 00 00 78 00 01 11
+	/* MCS Protect */
+	23 01 00 00 00 00 02 B0 00
+	/* CABC RGB full PWM */
+	29 01 00 00 00 00 02 FA 20
+	/* CABC PWM setting */
+	29 01 00 00 00 00 18 CE 5D 40 43 49 55 62 71 82 94 A8 B9 CB DB E9 FE FE FF 03 15 04 04 00 00
+	/* CABC UI mode setting */
+	29 01 00 00 00 00 07 B8 07 8B 19 08 0C 21
+	/* CABC Moving Image mode setting */
+	29 01 00 00 00 00 07 BA 07 7A 64 18 30 8D
+	/* Set Max Brightness */
+	15 01 00 00 00 00 02 51 BD
+	/* Set Minimum Brightness */
+	15 01 00 00 00 00 02 5E 00
+	/* BLU control */
+	15 01 00 00 00 00 02 53 2C
+	/* set defaut CABC to UI mode */
+	15 01 00 00 00 00 02 55 01
+	05 01 00 00 00 00 01 29];
+qcom,mdss-dsi-off-command = [
+	05 01 00 00 0a 00 01 28
+	05 01 00 00 55 00 01 10];
+
+qcom,mdss-dsi-cabc-ui-command = [15 01 00 00 00 00 02 55 01];
+qcom,mdss-dsi-cabc-mv-command = [15 01 00 00 00 00 02 55 03];
+qcom,mdss-dsi-cabc-dis-command = [15 01 00 00 00 00 02 55 00];
+qcom,mdss-dsi-hbm-on-command = [15 01 00 00 00 00 02 51 EB];
+qcom,mdss-dsi-hbm-off-command = [15 01 00 00 00 00 02 51 BD];
+
+qcom,mdss-dsi-on-command-state = "dsi_lp_mode";
+qcom,mdss-dsi-off-command-state = "dsi_lp_mode";
+qcom,mdss-dsi-h-sync-pulse = <0>;
+qcom,mdss-dsi-traffic-mode = "burst_mode";
+qcom,mdss-dsi-bllp-eof-power-mode;
+qcom,mdss-dsi-bllp-power-mode;
+qcom,mdss-dsi-lane-0-state;
+qcom,mdss-dsi-lane-1-state;
+qcom,mdss-dsi-lane-2-state;
+qcom,mdss-dsi-lane-3-state;
+qcom,mdss-dsi-panel-timings = [d6 32 22 00 60 66 26 36 28 03 04 00];
+qcom,mdss-dsi-t-clk-post = <0x3>;
+qcom,mdss-dsi-t-clk-pre = <0x28>;
+qcom,mdss-dsi-bl-min-level = <1>;
+qcom,mdss-dsi-bl-max-level = <255>;
+qcom,mdss-dsi-bl-on-defer-delay = <30>;
+qcom,mdss-dsi-dma-trigger = "trigger_sw";
+qcom,mdss-dsi-mdp-trigger = "none";
+qcom,mdss-dsi-bl-pmic-control-type = "bl_ctrl_wled";
+qcom,mdss-dsi-reset-sequence = <1 11>;
+qcom,mdss-dsi-lp11-init;
+qcom,mdss-dsi-tx-eot-append;
+
+qcom,mdss-dsi-te-pin-select = <1>;
+qcom,mdss-dsi-te-dcs-command = <1>;
+qcom,mdss-dsi-te-check-enable;
+qcom,mdss-dsi-te-using-te-pin;
+
+qcom,esd-check-enabled;
+qcom,mdss-dsi-panel-status-check-mode = "reg_read";
+qcom,mdss-dsi-panel-status-command = [06 01 00 01 00 00 01 0A];
+qcom,mdss-dsi-panel-status-command-state = "dsi_hs_mode";
+qcom,mdss-dsi-panel-status-value = <0x1c>;
+
+qcom,panel-supply-entries {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	qcom,panel-supply-entry@0 {
+		reg = <0>;
+		qcom,supply-name = "vddio_disp";
+		qcom,supply-min-voltage = <1800000>;
+		qcom,supply-max-voltage = <1800000>;
+		qcom,supply-enable-load = <300000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <0>;
+		qcom,supply-post-on-sleep = <0>;
+		qcom,supply-pre-off-sleep = <0>;
+		qcom,supply-post-off-sleep = <1>;
+	};
+
+	qcom,panel-supply-entry@1 {
+		reg = <1>;
+		qcom,supply-name = "vsp_disp";
+		qcom,supply-min-voltage = <4600000>;
+		qcom,supply-max-voltage = <6000000>;
+		qcom,supply-enable-load = <100000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <2>;
+		qcom,supply-post-on-sleep = <0>;
+		qcom,supply-pre-off-sleep = <0>;
+		qcom,supply-post-off-sleep = <0>;
+	};
+
+	qcom,panel-supply-entry@2 {
+		reg = <2>;
+		qcom,supply-name = "vsn_disp";
+		qcom,supply-min-voltage = <4600000>;
+		qcom,supply-max-voltage = <6000000>;
+		qcom,supply-enable-load = <100000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <0>;
+		qcom,supply-post-on-sleep = <2>;
+		qcom,supply-pre-off-sleep = <5>;
+		qcom,supply-post-off-sleep = <0>;
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd.dtsi
new file mode 100644
index 00000000000..eeba47fa549
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-mot-boe-546-1080p-cmd.dtsi
@@ -0,0 +1,17 @@
+/* Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+&mdss_mdp {
+	mot_boe_546_1080p_cmd_v0: qcom,mdss_dsi_mot_boe_546_1080p_cmd_v0 {
+		#include "dsi-panel-mot-boe-546-1080p-cmd-common.dtsi"
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts b/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
index feb6f953e2e..782ff39460d 100644
--- a/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
+++ b/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
@@ -24,6 +24,10 @@
 	qcom,dsi-pref-prim-pan = <&mot_tdi_546_1080p_cmd_v0>;
 };
 
+&mdss_mdp {
+	/delete-property/ mot_boe_546_1080p_cmd_v0;
+};
+
 &cci {
 	st,stmvl53l0
 	{
diff --git a/arch/arm/boot/dts/qcom/msm8952-athene.dtsi b/arch/arm/boot/dts/qcom/msm8952-athene.dtsi
index 0db6d9a955c..8b5dfeef02c 100644
--- a/arch/arm/boot/dts/qcom/msm8952-athene.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-athene.dtsi
@@ -14,6 +14,7 @@
 #include "msm8952-moto-common.dtsi"
 #include "msm8952-moto-dtv.dtsi"
 #include "dsi-panel-mot-lgd-546-1080p-cmd.dtsi"
+#include "dsi-panel-mot-boe-546-1080p-cmd.dtsi"
 
 / {
 	model = "Athene";
-- 
2.11.0

